### PR TITLE
feat(storage): update dual-region sample using CustomPlacementConfig

### LINKED ISSUE #2582: refactor(storage): Dual-Region Sample for Updated API
[Internal]
- For details and coordination: [go/gcs-dpe-dual-region-milestones-api-refactor](http://go/gcs-dpe-dual-region-milestones-api-refactor)
- Implementation Example: https://github.com/googleapis/nodejs-storage/pull/1977

### LINKED ISSUE #2644: storage/buckets: TestCreateBucketDualRegion failed
This test failed!

To configure my behavior, see [the Flaky Bot documentation](https://github.com/googleapis/repo-automation-bots/tree/main/packages/flakybot).

If I'm commenting on this issue too often, add the `flakybot: quiet` label and
I will stop commenting.

---

commit: 7edcf82edb8b1dd9e6e85269eb82b5433a6ec5bf
buildURL: [Build Status](https://source.cloud.google.com/results/invocations/b2484638-b3c5-40ab-b9d0-68aaa070ff0a), [Sponge](http://sponge2/b2484638-b3c5-40ab-b9d0-68aaa070ff0a)
status: failed
<details><summary>Test output</summary><br><pre>    buckets_test.go:108: location: got US, want location with regions US-WEST1 and US-CENTRAL1</pre></details>

### PR DESCRIPTION
**Pull request by cojenco**
Fixes #2582
Fixes #2644 

Tests pending https://github.com/googleapis/google-cloud-go/pull/6294 along with new library release

### CODE REVIEW

**Comment by enocom** [comment]
@cojenco does this PR need a new library version before it's ready for review?

**Comment by cojenco** [comment]
> @cojenco does this PR need a new library version before it's ready for review?



Yep this PR requires  a new library release once the feature changes are merged in, thanks!

File: storage/buckets/create_bucket_dual_region.go
Comment (tritone) [inline storage/buckets/create_bucket_dual_region.go:29]: We're supposed to limit function the args passed into samples to the minimum required for testing. Therefore, let's define location, region1 and region2 all in the func body rather than using args. See https://googlecloudplatform.github.io/samples-style-guide/#minimal-arguments

**Review by tritone — COMMENTED** [review]
One nit, otherwise LGTM once we merge the library feature and update.

File: storage/buckets/create_bucket_dual_region.go
Comment (cojenco) [reply storage/buckets/create_bucket_dual_region.go]: Thanks for linking! Updated to define vars inside the func body.

**Review by tritone — APPROVED** [review]

**Pull request merged**
